### ISSUE TITLE
Code inspection for unreachable Case expressions and statements

### ISSUE THREAD

**Issue by ThunderFrame** [rubberduck-vba/Rubberduck#1584]
> If testexpression matches any Case expressionlist clause, the statements following that Case statement run up to the next Case, Case Else, or End Select statement. Control then passes to the statement following End Select. If testexpression matches an expressionlist clause in more than one Case clause, only the statements following the first match run.

In large Select Case blocks, it's all too easy to inadvertently add Case statements that have intersections with prior Case blocks. (I found one in production code today)

That means that these blocks have unreachable case statements:

``` vb
Const x As String ="Bar"
Select Case x
  Case "Foo", "Bar"
    'Do FooBar
  Case "Foo"
    'Unreachable
  Case "Bar"
    'Unreachable
  Case "Foo"
    'Unreachable
End Select
```

``` vb
Const x as Long = 7

Select Case x
  Case 1 to 100
    'Do FooBar
  Case 50
    'Unreachable
End Select
```

We need to handle the special case of `Select Case True`

``` vb
Sub foo()
  Dim x As Long: x = InputBox("Enter a value for x")
  Dim y As Long: y = InputBox("Enter a value for y")
  Select Case True
    Case x = 8
      Stop
    Case y = 9
      Stop
    Case Else
      Stop
  End Select
End Sub
```

**Comment by comintern** [comment]
The `Select Case True` function highlights why parsing P-Code or compiled code helps to identify dead code:

```vba
Private Sub Foo()
    Select Case True
        Case Bar()
            Debug.Print "Bar"
        Case Baz()
            'Heuristically unreachable.
            Debug.Print "Baz"
    End Select
End Sub

Private Function Bar() As Boolean
    'Should trigger an inspection that it always returns true.
    Bar = True
End Function

Private Function Baz() As Boolean
    'Should trigger an inspection that it always returns false.
    Baz = False
End Function
```

**Comment by ThunderFrame** [comment]
Some Case handlers will be unreachable by virtue of their type. Here's a Boolean condition in the `Select Case` statement.

```vb
  Select Case VBA.Rnd > 0.5
    Case True
      Stop
    Case False
      Stop
    Case Else 'Unreachable
      Stop
  End Select
```

While some Case handlers will be unreachable by virtue of implicit type casting:
```vb
  Select Case VBA.Rnd > 0.5
    Case -5 'Implicitly True
      Stop
    Case -4 'Unreachable
      Stop
    Case Else
      Stop
  End Select
```

**Comment by bclothier** [comment]
We can consider this closed by PR #3868 -- any issues can be opened separately.
